cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/network/util/ConnectionExt.kt

```diff
@@ -0,0 +1,30 @@
+package com.sourcepoint.cmplibrary.data.network.util
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+
+/**
+ * Extension function that checks whether or not the device is connected to the Internet
+ */
+fun Context.isInternetConnected(): Boolean {
```

Comment:
This is good!
This should be internal and used from the ConsentManager

Reply:
I have 2 questions here:
1. is `ConsentManager` a part of data layer or not?
2. what is the purpose of using this check specifically in `ConsentManager`?